zerver/tests/test_subs.py

```diff
@@ -456,6 +456,26 @@ def test_deactivate_stream_removes_default_stream(self) -> None:
         do_deactivate_stream(stream)
         self.assertEqual(0, DefaultStream.objects.filter(stream_id=stream.id).count())
 
+    def test_deactivate_stream_removes_stream_from_default_stream_groups(self) -> None:
+        realm = get_realm('zulip')
+        streams = []
+        for stream_name in ["stream1", "stream2", "stream3"]:
+            stream = ensure_stream(realm, stream_name)
+            streams.append(stream)
+
+        def get_streams(group: DefaultStreamGroup) -> List[Stream]:
+            return list(group.streams.all().order_by('name'))
+
+        group_name = "group1"
+        description = "This is group1"
+        do_create_default_stream_group(realm, group_name, description, streams)
+        default_stream_groups = get_default_stream_groups(realm)
+        self.assertEqual(get_streams(default_stream_groups[0]), streams)
+
+        do_deactivate_stream(streams[2])
+        remaining_streams = streams[0:2]
+        self.assertEqual(get_streams(default_stream_groups[0]), remaining_streams)
```

Comment:
I don't like this test; it's too fragile with all the list-slicing. 

Maybe start with 
```
stream_to_remove = ["stream1"]
streams_to_keep = ["stream2"]
for stream_name in streams_to_keep + stream_to_remove:
    ...
```